LibraryInstrumentationGuidance.md

```diff
@@ -1,53 +1,115 @@
+* **Minimal performance impact**. In absence of listener, instrumentation has neglectable impact on the performance. When listener is present, it is able to sample events to minimize the reasonable impact instrumentation may have.
+* **Baked into the platform**. To make sure everything could be instrumented (e.g. .NET HttpClient or SqlClient), instrumentation primitives are part of .NET
+* **General purpose**. Could be consumed by any tracing/monitoring/profiling tool and the tool can choose level of details it gives its users
 
 ## Diagnostic Source and Activities
 
 [Diagnostic Source][DiagnosticSourceGuide] is a simple module that allows code to be instrumented for production-time logging of rich data payloads for consumption within the process that was instrumented. At runtime, consumers can dynamically discover data sources and subscribe to the ones of interest.
 
-[Activity][ActivityGuide] is a class that allows storing and accessing diagnostics context and consuming it with logging system.
+[Activity][ActivityGuide] is a class that allows storing and accessing diagnostics context and consuming it with logging system. Activity flows with async calls and anything can request Activity.Current and correlate the trace/log with it.
 
 Both Diagnostic Source and Activity have been used to instrument [System.Net.Http][SystemNetHttp] and [Microsoft.AspNetCore.Hosting][MicrosoftAspNetCoreHosting].
-
 More recently two new libraries were instrumented and that work was the basis for this guidance. These libraries are client SDKs for [Azure Event Hubs][MicrosoftEventHubs] and [Azure Service Bus][MicrosoftServiceBus], both of which support high throughput scenarios.
 
-[This document][DiagnosticSourceActivityHowto] goes into more details on how to efficiently use Diagnostic Source.
-
 ## What should be instrumented
 
 The goal of instrumentation is to give the users the visibility to how particular operations are being performed inside the library. This information can be later used to diagnose performance or issues. It is up to the library authors to identify operations that the library performs and are worthy of monitoring. These operations can match the exposed API but also cover more specific internal logic (like outgoing service calls, retries, locking, cache utilization, awaiting system events, etc.). This way the users can get a good understanding of what's going on under the hood when they need it. 
 
-## Instrumentation 
+## Bare Minimum Instrumentation 
 
-In the simplest case, the operation that is being monitored has to be wrapped by an activity. However in order to minimize performance impact the activity should be only created if there is any listener waiting for it.
+In the simplest case, the operation that is being monitored has to be wrapped by an activity. 
 
 ```csharp
+    // create DiagnosticListener with unique name
     static DiagnosticListener source = new DiagnosticListener("Example.MyLibrary");
```

Comment:
I would change Example.MyLibrary, to something like 'NameSpace.ComponentName'  to give people guidance on what a good name looks like